src/core/ext/filters/client_channel/resolver/dns/c_ares/dns_resolver_ares.cc

```diff
@@ -320,21 +319,35 @@ void AresDnsResolver::OnResolvedLocked(grpc_error_handle error) {
     GRPC_ERROR_UNREF(error);
     return;
   }
+  // TODO(roth): Change logic to be able to report failures for addresses
```

Comment:
One thing that might be unexpected here is that this code will only relay service config errors if we successfully resolved one but failed to parse it.

However, it won't relay errors for failing to resolve a service config.

Does this TODO capture that?

Reply:
Yeah, that's part of this TODO.

I agree that the current behavior is sub-optimal, but I think it's the same behavior it's had all along, so this PR doesn't really change anything with respect to that.  There's also a related TODO [here](https://github.com/grpc/grpc/blob/9722651156ddf97e8b90d991b379a4692c5a344e/src/core/ext/filters/client_channel/resolver/dns/c_ares/grpc_ares_wrapper.cc#L593) about being able to return a service config with an empty address list; I think that's something else we should fix at the same time.  Basically, the idea is that the address lookup and the TXT lookup should be completely independent of each other.